### PR TITLE
[4.0][WIP][RFC] Child templates part 2: com_templates

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
This is still work in progress read the comments bellow

### CODE REVIEW

**Comment by dgrammatiko** [comment]
So, as I promised I am going to work on finalising the child templates proposal.
The approach here is the one that was proposed by @fedik https://github.com/joomla/joomla-cms/pull/30149#issuecomment-663104018

> Move button "Create Child" to a template style form, and rename to "Override template files" (or something similar)

Screenshot 2020-08-13 at 12 03 24

> This "new template" should not be visible in "templates list", only via template style. (this need to avoid confusing between all these template things)

Screenshot 2020-08-13 at 12 04 34


So far this PR is based on mild changes on the current workflow of com_templates and if this is what people are ok then the code should be ready in the coming days for testing. But since I'm spending some time reworking basically most parts of the component a better workflow, more intuitive could be a better approach here. Let me explain: 
- right now we have 2 list views (templates & styles) that in the component level are not really connected
- we could merge these 2 list views in one and have a list of the templates (as is right now) with an additional accordion like component so when someone clicks on the template the styles are revealed (and clicking on a style the toolbar reveals the appropriate buttons)
- also there was some work already done by @kawshar 's team which maybe a good idea to bring it in (I mean if they're ok with it)


Anyways, suggestions, ideas, feedback is welcome here...

**Comment by kawshar** [comment]
You are welcome to adopt anything from our concept. For Joomla :)

About the multiple views, we can easily merge both without breaking anything.

**Comment by brianteeman** [comment]
Definitely in favour of merging the template views.

File: administrator/language/en-GB/com_templates.ini [hunk 1]
```diff
@@ -37,6 +37,8 @@ COM_TEMPLATES_CONFIG_SOURCE_LABEL="Valid Source Formats"
 COM_TEMPLATES_CONFIG_UPLOAD_LABEL="Max. Upload Size (MB)"
 COM_TEMPLATES_CONFIGURATION="Template: Options"
 COM_TEMPLATES_COPY_SUCCESS="New template called %s was installed."
+JTOOLBAR_CREATE_CHILD="Duplicate as child"
+JTOOLBAR_OVERRIDE_CHILD="Manage overrides"
```
Comment (brianteeman) [inline administrator/language/en-GB/com_templates.ini]: As per the style guide https://developer.joomla.org/en-gb-user-interface-text-guidelines/capitalisation.html

```suggestion
JTOOLBAR_CREATE_CHILD="Duplicate as Child"
JTOOLBAR_OVERRIDE_CHILD="Manage Overrides"
```

**Comment by dgrammatiko** [comment]
> Definitely in favour of merging the template views.

Just a comment here about merging the 2 views together: there are quite some issues that need to be addressed:
- Extremely hard to get thee filters working with both the templates and their styles
- The toolbar might need a bit of custom code to follow selections (template/style)
- pagination
- probably a few more that I can't even think about right now

That said combining the views is really not that hard (code wise) but we need at least for the first step some sort of design, eg and unstyled, not really functional page was just few lines:
Aug-14-2020 14-51-24

**Comment by dgrammatiko** [comment]
moving to https://github.com/joomla/joomla-cms/pull/32896

**Pull request closed without merging**